cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java

```diff
@@ -311,7 +312,12 @@ private KafkaMirrorMakerConsumerConfiguration getConsumerConfiguration() {
         KafkaMirrorMakerConsumerConfiguration config = new KafkaMirrorMakerConsumerConfiguration(reconciliation, consumer.getConfig().entrySet());
 
         if (tracing != null) {
-            config.setConfigOption("interceptor.classes", "io.opentracing.contrib.kafka.TracingConsumerInterceptor");
+            // using hard-coded string because of the -Werror flag
+            if ("jaeger".equals(tracing.getType())) {
```

Comment:
Hmm, I wonder if using switch could resolve your problem with `-Werror` and then use just
```java
switch (tracing.getType) { 
    case OpenTelemetryTracing.TYPE_OPENTELEMETRY:
       ....
    case JaegerTracing.TYPE_JAEGER:
      ....
```

Reply:
Actually I was not having the problem with the same code in the `KafkaMirrorMaker2AssemblyOperator` then I noticed that there was a `@SuppressWarnings("deprecation")` there already. I will use the same here.